lib/plugins/aws/provider/awsProvider.js

```diff
@@ -410,15 +410,15 @@ class AwsProvider {
 
   getProfileSourceValue() {
     const values = this.getValues(this, [
-      ['options', 'profile'],
+      ['options', 'aws-profile'],
```

Comment:
Is this option being renamed?  This strikes me as a breaking change.

Reply:
The option remains `profile` according to https://serverless.com/framework/docs/providers/aws/cli-reference/config-credentials/ though I did find https://serverless.com/framework/docs/providers/aws/guide/credentials#using-the-aws-profile-option which suggests that `aws-profile` is correct.  However, that's never been supported so I'd suggest this is a documentation bug.

Reply:
Yeah. I wasn't 100% sure about this. I was only familiar with the later option, which is global. I see that it's an option specifically on the `config` subcommand 😕 

I've got an idea for how to resolve this. (the real issue is that `options.profile` shouldn't be checked while doing variable substitution because it shadows the standard way of setting variables from options)

Reply:
Can you please explain how that's the real issue?  I'm a bit confused about the assertion and change.

That part of the code is only about prioritizing the resolution/variables of specific attributes to avoid dependency related issues in non-variable usage (e.g. use in the aws-sdk).

My expectation, taking the assumption that their is some improper handling of `--aws-profile`, would be something like you would add the `--aws-profile` option rather than contingently using the existing option.

Reply:
discussed in slack and completely changed anyway